File: Python/1.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dic = dict()
        for index in range(len(nums)):
            diff = target - nums[index]
            if diff in dic :
                return [index, dic[diff]]
            else:
                dic[nums[index]] = index

File: C++/1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> answer;
        int len = nums.size();
        unordered_map<int, int> map;
        unordered_map<int, int>::iterator pos;
        for(int i = 0; i < len; i++) {
            pos = map.find(nums[i]);
            if(pos != map.end()){
                answer.push_back(pos->second);
                answer.push_back(i);
                break;
            }else{
                map[target - nums[i]] = i;
            }
        }
        return answer;
    }
};
